File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []
        ans.append(1)

        pref = 1
        for i in range(len(nums)-1):
            pref *= nums[i]
            ans.append(pref)
        ans.reverse()
        nums.reverse()
        post = 1
        for i in range(len(ans)):
            ans[i] *= post
            post *= nums[i]
        ans.reverse()
        return ans
        

File: Data Structures & Algorithms/missing-number/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    missingNumber(nums) {
        let ans = nums.length;
        for(let i = 0; i < nums.length; i++){
            ans = ans ^ i;
            ans = ans ^ nums[i];
        }

        return ans;
    }
}
